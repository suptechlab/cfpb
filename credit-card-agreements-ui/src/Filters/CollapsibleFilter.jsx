import './CollapsibleFilter.less';
import PropTypes from 'prop-types'
import React from 'react';

export default class CollapsibleFilter extends React.Component {
  constructor( props ) {
    super( props );
    this.state = {
      showChildren: props.showChildren
    };

    // This binding is necessary to make `this` work in the callback
    // https://facebook.github.io/react/docs/handling-events.html
    this._toggleChildDisplay = this._toggleChildDisplay.bind( this );
  }

  _toggleChildDisplay() {
    this.setState( {
      showChildren: !this.state.showChildren
    } );
  }

  componentWillReceiveProps( nextProps ) {
    this.setState( {
      showChildren: nextProps.showChildren
    } );
  }

  render() {
    let composeClasses = 'o-expandable'
    if ( this.props.className ) {
      composeClasses += ' ' + this.props.className;
    }

    const buttonClasses = 'a-btn a-btn__link o-expandable_cue '

    const opened =
      <button className={ buttonClasses + 'o-expandable_cue-close' }
              onClick={this._toggleChildDisplay}>
        Hide
        <span className="cf-icon cf-icon-minus-round"></span>
      </button>

    const closed =
      <button className={ buttonClasses + 'o-expandable_cue-open' }
              onClick={this._toggleChildDisplay}>
        Show
        <span className="cf-icon cf-icon-plus-round"></span>
      </button>

    return (
      <section>
        <div className="o-expandable_header o-expandable_target">
          <h4 className="o-expandable_header-left o-expandable_label">
              {this.props.title}
          </h4>
        </div>
        { this.props.desc ? '<p>' + this.props.desc + '</p>' : '' }
        { this.state.showChildren ? this.props.children : null }
      </section>
    );
  }
}

CollapsibleFilter.propTypes = {
  showChildren: PropTypes.bool
}

CollapsibleFilter.defaultProps = {
  showChildren: true
}
