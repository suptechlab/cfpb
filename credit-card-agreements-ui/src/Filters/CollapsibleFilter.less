@import (less) "../base.less";

.o-expandable {
  h4 {
    max-width: ~"calc(100% - 55px)";
    box-sizing: border-box;
    padding-right: @gutter-minimum;
  }

  .o-expandable_cue {
    &.a-btn__link {
      border: 0;

      font-size: @size-v;
    }

    .cf-icon {
      padding-left: @gutter-minimum;
    }
  }

  p {
    color: @gray;
    font-size: 16px;
  }
}