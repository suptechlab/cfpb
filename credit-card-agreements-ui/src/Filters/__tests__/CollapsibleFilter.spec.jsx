import React from 'react';
import { mount } from 'enzyme';
import CollapsibleFilter from '../CollapsibleFilter';
import renderer from 'react-test-renderer';

describe('initial state', () => {
  it('renders without crashing', () => {
    const target = renderer.create(
      <CollapsibleFilter title="foo" desc="bar" />
    );

    let tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
