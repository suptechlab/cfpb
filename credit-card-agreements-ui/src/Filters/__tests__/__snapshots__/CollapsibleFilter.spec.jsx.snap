// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial state renders without crashing 1`] = `
<section>
  <div
    className="o-expandable_header o-expandable_target"
  >
    <h4
      className="o-expandable_header-left o-expandable_label"
    >
      foo
    </h4>
  </div>
  &lt;p&gt;bar&lt;/p&gt;
</section>
`;
